tests/test_commands.py

```diff
@@ -234,6 +233,62 @@ def test_nickserv_and_memoserv(alice, bob, wait_until):
     wait_until(lambda: "send Bob hello there\n" in bob.text())
 
 
+@pytest.mark.skipif(
+    os.environ["IRC_SERVER"] == "hircd",
+    reason="hircd doesn't support away notifications",
+)
+def test_away_status(alice, bob, wait_until):
+    switch_to_channel_view(bob, "#autojoin")
+
+    alice.entry.insert("end", "/away foo bar baz")
+    alice.on_enter_pressed()
+
+    # Server view
+    wait_until(lambda: "You have been marked as being away\n" in alice.text())
+
+    switch_to_channel_view(alice, "#autojoin")
+
+    # Channel view
+    wait_until(lambda: "You have been marked as being away\n" in alice.text())
+
+    assert "away" in alice.get_current_view().userlist.treeview.item("Alice")["tags"]
+
+    # TODO: Get this to work
+    # assert "away" in bob.get_current_view().userlist.treeview.item("Alice"))
```

Comment:
The problem with this assert is that commands aren't immediate. The commands go through a send queue and a couple different socket connections, and all that takes time.

But what matters is that Alice will eventually show up as being away. That's what `wait_until()` does:

```suggestion
    wait_until(lambda: "away" in bob.get_current_view().userlist.treeview.item("Alice")["tags"])
```

The `wait_until()` call will error if `"away"` doesn't appear in Alice's tags within 5 seconds. The test passes on my computer with this change applied.